app/models/game.rb

```diff
@@ -36,4 +36,27 @@ def other_player(player)
   def pass_turn(player)
     update_attributes(active_player_id: other_player(player))
   end
+
+  def stalemate?
+    king_not_in_check? && available_move?
+  end
+
+  def king_not_in_check?
+    king = pieces.find_by(type: 'King', color: current_color)
+    return true unless king.in_check?
+    false
+  end
+
+  def available_move?
+    all_piece = pieces.where(in_game: true)
```

Comment:
I don't think it's really been set up yet(I think we need to add it to populate_the_board or whatever it's called) but you  could potentially just only grab the current player's pieces since you only have to check if the current player has any valid moves, which should make things a little quicker by taking out any unnecessary calculations.